File: src/main/java/liqp/filters/Remove.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Remove extends Filter {

    /*
     * remove(input, string)
     *
     * remove a substring
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        String original = super.asString(value);

        Object needle = super.get(0, params);

        if (needle == null) {
            throw new RuntimeException("invalid pattern: " + needle);
        }

        return original.replace(String.valueOf(needle), "");
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Remove.kt
package liqp.filter

import liqp.context.LContext

class Remove : LFilter() {

  /**
   * remove(input, string)
   *
   * remove a substring
   */
  override fun onFilterAction(params: FilterParams, value: Any?, chain: FilterChainPointer, context: LContext): Any? {
    val original = context.asString(value)
    val needle:Any = params[0] ?: throw RuntimeException("invalid pattern: " + params[0])

    return original?.replace(needle.toString(), "")
  }
}
